code/game/objects/items/reagent_containers/food/snacks.dm

```diff
@@ -2792,3 +2792,76 @@
 		if("spiced apples", "chocolate brownie", "sugar cookie", "choco bar", "crayon")
 			icon_state = "dessert"
 			list_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/sugar = 1)
+
+
+/obj/item/reagent_containers/food/snacks/lollipop
+	name = "lollipop"
+	desc = "A delicious lollipop."
+	icon = 'icons/obj/items/lollipop.dmi'
+	icon_state = "lollipop_stick"
+	item_state = "lollipop_stick"
+	flags_equip_slot = ITEM_SLOT_MASK
+	list_reagents = list(/datum/reagent/consumable/nutriment = 10, /datum/reagent/consumable/sugar = 15)
+	var/mutable_appearance/head
+	var/headcolor = rgb(0, 0, 0)
+	var/succ_dur = 180
+	var/succ_int = 100
+	var/next_succ = 0
+	tastes = list("candy" = 1)
+ 
+/obj/item/reagent_containers/food/snacks/lollipop/Initialize()
+	. = ..()
+	head = mutable_appearance('icons/obj/items/lollipop.dmi', "lollipop_head")
+	change_head_color(rgb(rand(0, 255), rand(0, 255), rand(0, 255)))
+ 
+//makes lollipops actually wearable as masks and still edible the old fashioned way.
+/obj/item/reagent_containers/food/snacks/lollipop/proc/handle_reagents()
+	var/mob/living/carbon/C = loc
+	if(src == C.wear_mask) // if it's in the human/monkey mouth, transfer reagents to the mob
+		var/fraction = min(REAGENTS_METABOLISM/reagents.total_volume, 1)
+		reagents.reaction(C, INGEST, fraction)
+		if(!reagents.trans_to(C, REAGENTS_METABOLISM))
+			reagents.remove_any(REAGENTS_METABOLISM)
+		return
+	reagents.remove_any(REAGENTS_METABOLISM)
+ 
+/obj/item/reagent_containers/food/snacks/lollipop/process()
+	if(!iscarbon(loc))
+		stack_trace("lollipop processing outside of a carbon loc")
+		return PROCESS_KILL
+	if(!reagents)
+		stack_trace("lollipop processing without a reagents datum")
+		return PROCESS_KILL
+	if(succ_dur < 1)
+		qdel(src)
+		return
+	succ_dur--
+	if((reagents.total_volume) && (next_succ <= world.time))
+		handle_reagents()
+		next_succ = world.time + succ_int
+ 
+/obj/item/reagent_containers/food/snacks/lollipop/equipped(mob/user, slot)
```

Comment:
replace this with `COMSIG_ITEM_EQUIPPED_TO_SLOT`
and also stop processing on `COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT` and `COMSIG_ITEM_DROPPED`